// parallel-web/resources/beta/task-run.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class TaskRun extends APIResource {
    /**
     * Initiates a task run.
     *
     * Returns immediately with a run object in status 'queued'.
     *
     * Beta features can be enabled by setting the 'parallel-beta' header.
     */
    create(params, options) {
        const { betas, ...body } = params;
        return this._client.post('/v1/tasks/runs?beta=true', {
            body,
            ...options,
            headers: buildHeaders([
                { ...(betas?.toString() != null ? { 'parallel-beta': betas?.toString() } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Streams events for a task run.
     *
     * Returns a stream of events showing progress updates and state changes for the
     * task run.
     *
     * For task runs that did not have enable_events set to true during creation, the
     * frequency of events will be reduced.
     */
    events(runID, options) {
        return this._client.get(path `/v1beta/tasks/runs/${runID}/events`, {
            ...options,
            headers: buildHeaders([{ Accept: 'text/event-stream' }, options?.headers]),
            stream: true,
        });
    }
    /**
     * Retrieves a run result by run_id, blocking until the run is completed.
     */
    result(runID, params = {}, options) {
        const { betas, ...query } = params ?? {};
        return this._client.get(path `/v1/tasks/runs/${runID}/result?beta=true`, {
            query,
            ...options,
            headers: buildHeaders([
                { ...(betas?.toString() != null ? { 'parallel-beta': betas?.toString() } : undefined) },
                options?.headers,
            ]),
        });
    }
}
